src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -47,36 +50,52 @@ def work_horse(job_parameters: dict) -> Tuple[pd.DataFrame, Dict[str, pd.DataFra
     logger.info(f"Starting job: {job_parameters}")
 
     try:
-        sim = setup_sim(job_parameters)
-
-        start_time = pd.Timestamp(**sim.configuration.time.start.to_dict())
-        end_time = pd.Timestamp(**sim.configuration.time.end.to_dict())
-        step_size = pd.Timedelta(days=sim.configuration.time.step_size)
-        num_steps = int(math.ceil((end_time - start_time) / step_size))
-
         start_snapshot = CounterSnapshot()
         event = {"start": time()}  # timestamps of application events
         logger.info("Beginning simulation setup.")
-        sim.setup()
-        event["simulant_initialization_start"] = time()
-        exec_time = {
-            "setup_minutes": (event["simulant_initialization_start"] - event["start"]) / 60
-        }  # execution event
-        logger.info(
-            f'Simulation setup completed in {exec_time["setup_minutes"]:.3f} minutes.'
-        )
-
-        sim.initialize_simulants()
-        event["simulation_start"] = time()
-        exec_time["simulant_initialization_minutes"] = (
-            event["simulation_start"] - event["simulant_initialization_start"]
-        ) / 60
-        logger.info(
-            f'Simulant initialization completed in {exec_time["simulant_initialization_minutes"]:.3f} minutes.'
-        )
-
-        logger.info(f"Starting main simulation loop with {num_steps} time steps")
-        sim.run()
+        backup = get_backup(job_parameters)
+        if backup:
+            logger.info(f"Restarting simulation from saved backup")
+            sim = backup
+            event["simulation_start"] = time()
+            exec_time = {
+                "setup_minutes": (event["simulation_start"] - event["start"]) / 60
+            }  # execution event
+            logger.info(
+                f'Simulation setup from backup completed in {exec_time["setup_minutes"]:.3f} minutes.'
+            )
+            logger.info(f"Starting main simulation loop")
+        else:
```

Comment:
Everything in this else block is the same as before, correct?

Reply:
yes, I've moved a couple things around now but there is no functional change.